format/bson/bson.go

```diff
@@ -79,28 +86,38 @@ func decodeBSONDocument(d *decode.D) {
 					case elementTypeArray:
 						d.FieldStruct("value", decodeBSONDocument)
 					case elementTypeBinary:
-						length := d.FieldU32("length")
+						length := d.FieldS32("length")
 						d.FieldU8("subtype")
-						d.FieldRawLen("value", int64(length)*8)
+						d.FieldRawLen("value", length*8)
 					case elementTypeUndefined:
 						//deprecated
 					case elementTypeObjectID:
 						d.FieldRawLen("value", 12*8)
 					case elementTypeBoolean:
 						d.FieldU8("value")
-					case elementTypeDatatime:
-						d.FieldS32("value")
+					case elementTypeDatetime:
+						d.FieldS64("value")
 					case elementTypeNull:
 						d.FieldValueAny("value", nil)
 					case elementTypeRegexp:
 						d.FieldUTF8Null("value")
 						d.FieldUTF8Null("options")
+					case elementTypeJavaScript:
+						length := d.FieldS32("length")
+						d.FieldUTF8NullFixedLen("value", int(length))
 					case elementTypeInt32:
 						d.FieldS32("value")
 					case elementTypeTimestamp:
 						d.FieldU64("value")
 					case elementTypeInt64:
 						d.FieldS64("value")
+					case elementTypeDecimal128:
+						// TODO: Parse the IEEE 754 decimal128 value.
+						d.FieldRawLen("value", 128)
```

Comment:
Would it make sense to decode but store it as float64 or just confusing? one could add `big.Rat` support to the decode API but not that useful as gojq dont' support it. Something to think about.

Reply:
I'd like to add support for decoding decimal128 values so the logical value can be displayed. I'm not sure yet how to integrate it into the `decode` API or the rest of `fq`, tho. Converting it to a float64 would be simple, but could be confusing because it might lose precision during conversion. I'll investigate further and see if I can come up with a good solution.

Reply:
To clarify, I'm not currently planning to add decimal128 decoding support to this PR. Do you think it's OK to add decimal128 support in a subsequent PR?

Reply:
No worries, and sorry that what i meant also, leave it for future improvement outside this PR. I can help out with adding `big.Rat` support if needed. One hacky way if you want to just show the value for now could be to assign it as a text description, hmm not great either :) but lets merge this without more feature and iterate on it if needed.